### PR TITLE
gui: Show a warning when authentication is not configured but listening on localhost (fixes #4703)

### LINKED ISSUE #4703: Suggest setting up auth on initial startup
We could use the persistent notification thingy to display a notice that explains that a username and password should be set when on a shared computer. I don't want to make this a hard requirement so a notification that is displayed until acknowledged would be fine with me. 

Relates to #3357.

### PR DESCRIPTION
**Pull request by e-sch**
### Purpose

This shows a panel to warn the user that the UI can be accessed by other users on the same machine. It will only show when listening on localhost and remote access is not possible as there is already a warning for that.

The panel can be dismissed and it will not show up again across reloads and restarts.

I wasn't sure whether I should use the panel-warning or panel-info CSS class, can someone make a suggestion?

### Testing
- Set `STGUIADDRESS` to `127.0.0.1:8384` and disable auth -> "Warning!" panel appears
- Set `STGUIADDRESS` to `0.0.0.0:8384` and disable auth -> "Danger!" panel appears (can not be dismissed)
- Click `Do not show again` and reload page / restart Syncthing -> warning panel does not appear
- Set `STGUIADDRESS` to `127.0.0.1:8384` and setup auth -> warning panel does not appear

### Screenshots
2019-10-27-000928_1187x618_scrot

### CODE REVIEW

File: lib/config/config_test.go [hunk 1]
```diff
@@ -68,6 +68,7 @@ func TestDefaultValues(t *testing.T) {
 		OverwriteRemoteDevNames: false,
 		TempIndexMinBlocks:      10,
 		UnackedNotificationIDs:  []string{},
+		DismissedPanels:         []string{},
```
Comment (imsodin) [inline lib/config/config_test.go]: I think you could just use `UnackedNotificationIDs` for this. It's admittedly not it's original purpose, but I'd personally still rather stretch its scope than add a new config field just for this panel.

File: gui/default/index.html [hunk 2]
```diff
@@ -122,6 +122,39 @@ <h3 class="panel-title">
...
+              <div class="panel-icon">
+                <span class="fas fa-exclamation-circle"></span>
+              </div>
+              <span translate>Warning!</span>
+            </h3>
+          </div>
+          <div class="panel-body">
+            <p>
+            <span translate>If this is a shared computer, other users might be able to access the admin interface.</span>
+            <span translate>Please set a GUI Authentication User and Password in the Settings dialog.</span>
```
Comment (imsodin) [inline gui/default/index.html]: I wouldn't capitalize user, password and settings. And I'd word it something like "You can set a user and password in the GUI tab of the settings dialog to restrict access".

File: gui/default/index.html [hunk 3]
```diff
@@ -122,6 +122,39 @@ <h3 class="panel-title">
...
+            <h3 class="panel-title">
+              <div class="panel-icon">
+                <span class="fas fa-exclamation-circle"></span>
+              </div>
+              <span translate>Warning!</span>
+            </h3>
+          </div>
+          <div class="panel-body">
+            <p>
+            <span translate>If this is a shared computer, other users might be able to access the admin interface.</span>
```
Comment (imsodin) [inline gui/default/index.html]: Is this intentionally conditional? Otherwise just "other users can access".

File: lib/config/config_test.go (hunk 1 above)
Comment (AudriusButkevicius) [reply lib/config/config_test.go]: Or use the same mechanism that UnackedNotificationIDs  uses to show a one-off modal like we do for other stuff, just change the display condition to have the additional constraints on addresses and other stuff.

**Comment by st-review** [comment]
:robot: *beep boop*

I'm going to close this pull request as it has been idle for more than 90 days.

This is not a rejection, merely a removal from the list of active pull requests that is periodically reviewed by humans. The pull request can be reopened when there is new activity, and merged once any remaining issues are resolved.

**Pull request closed without merging**